codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +404,237 @@ public TextHeaders clear() {
+            DefaultTextHeaders.super.setObject(name, values);
+            return DefaultTextHeaders.this;
+        }
+    }
+
+    /**
+     * Will add multiple values for the same header as single header with a comma separated list of values.
+     *
+     * Please refer to section <a href="https://tools.ietf.org/html/rfc7230#section-3.2.2">3.2.2 Field Order</a>
+     * of RFC-7230 for details.
+     */
+    private final class SingleHeaderValuesComposer implements ValuesComposer {
+
+        private static final String COMMA = ",";
+        private final ValueConverter<CharSequence> valueConverter;
+
+        SingleHeaderValuesComposer() {
+            valueConverter = valueConverter();
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence value) {
+            CharSequence currentValue = DefaultTextHeaders.super.get(name);
+            if (currentValue == null) {
+                DefaultTextHeaders.super.add(name, value);
+            } else {
+                DefaultTextHeaders.super.set(name, commaSeparate(currentValue, value));
+            }
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence... values) {
+            return addConverted(name, commaSeparate(values));
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, Iterable<? extends CharSequence> values) {
+            return addConverted(name, commaSeparate(values));
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Iterable<?> values) {
+            return addConverted(name, convertAndCommaSeparate(values));
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Object... values) {
+            return addConverted(name, convertAndCommaSeparate(values));
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.set(name, commaSeparate(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.set(name, convertAndCommaSeparate(values));
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.set(name, convertAndCommaSeparate(values));
+            return DefaultTextHeaders.this;
+        }
+
+        private TextHeaders addConverted(CharSequence name, CharSequence convertedValue) {
```

Comment:
Oops I moved my comment to https://github.com/netty/netty/pull/3271#discussion-diff-22869993.  Figured it was a better spot to be removed when you updated the code.